File: clients/src/main/java/eu/getsoftware/hotelico/hotel/clients/chat/BasicDTO.java
package eu.getsoftware.hotelico.hotel.clients.chat;

import java.io.Serializable;

/**
 * Created by Eugen on 08.02.2016.
 */
abstract class BasicDTO implements Serializable
{
	Long id = 0L;
	
	Long initId = 0L;
	
	Boolean active = true;
	
	String dtoType = this.getClass().getSimpleName().toLowerCase();
	
	BasicDTO(){
		
	};
	
	BasicDTO(Long initId) {
        this.initId = initId;
    }
}

File: chat/src/main/java/eu/getsoftware/hotelico/hotel/chat/dto/BasicDTO.kt
package eu.getsoftware.hotelico.hotel.chat.dto

import java.io.Serializable

/**
 * Created by Eugen on 08.02.2016.
 */
abstract class BasicDTO: Serializable
{
    var id : Long = 0L

    open var initId : Long = 0L

    var active: Boolean = true

    val dtoType: String = this.javaClass.simpleName.toLowerCase()

    constructor()

    constructor(initId: Long) {
        this.initId = initId
    }
}
